serialization.rst

```diff
@@ -0,0 +1,308 @@
+===============================
+Persistence in Hippo and Phongo
+===============================
+
+This document discusses the methods how compound structures (documents,
+arrays, objects) are persisted through the drivers. And how they are brought
+back into PHP land.
+
+Serialization to BSON
+=====================
+
+Arrays
+------
+
+If an array is a *packed array* — i.e. the keys start at 0 and are sequential
+without gaps: **BSON array**. 
+
+If the array is not packed — i.e. having associative (string) keys, the keys
+don't start at 0, or when they are gaps:: **BSON object**
+
+Examples
+~~~~~~~~
+
+These serialize as a BSON array::
+
+  [ 8, 5, 2, 3 ] => [ 8, 5, 2, 3 ]
+  [ 0 => 4, 1 => 9 ] => [ 4, 9 ]
+
+
+These serialize as a BSON document::
+
+  [ 0 => 1, 2 => 8, 3 => 12 ] => { "0" : 1, "2" : 8, "3" : 12 }
+  [ "foo" => 42 ] => { "foo" : 42 }
+  [ 1 => 9, 0 => 10 ] => { "1" : 9, "0" => 10 }
+
+Objects
+-------
+
+If an object is of the *stdClass*, serialize as a **BSON document**.
+
+If an object is of any other class, without implementing any special
+interface, serialize as a **BSON document**. Keep only *public* properties,
+and ignore *protected* and *private* properties.
```

Comment:
I agree with "ignore *protected* and *private* properties" and will note that this should satisfy [PHP-1457](https://jira.mongodb.org/browse/PHP-1457).